aeronet_raster/test/unit/test_utils.py

```diff
@@ -20,3 +28,73 @@ def test_utm_zone_from_projected():
                         Affine(1.0, 0, 1000.0, 0, -1.0, -1000.0),
                         (100, 100)) == CRS.from_epsg(32731)
 
+
+def _run_collection_processor(height=3000, width=400, gen_mode='random',
+                              sample_size=(1500, 1500), bound=100, bound_mode=None):
+    def read_bc(path, bands):
+        bands = parse_directory(path, bands)
+        return BandCollection(bands)
+
+    def processing_fn(sample):
+        out = sample.numpy()
+        return out
+
+    input_channels = ['input']
+    output_labels = ['output']
+
+    tempdir = TemporaryDirectory()
+    path = Path(tempdir.name)
+    filename = path / 'input.tif'
+    create_tiff_file(filename, width, height, mode=gen_mode, count=1)
+
+    bc = read_bc(path, input_channels)
+
+    if bound_mode is None:
+        cp = CollectionProcessor(
+            input_channels=input_channels,
+            output_labels=output_labels,
+            processing_fn=processing_fn,
+            sample_size=sample_size,
+            bound=bound,
+            verbose=True
+        )
+    else:
+        cp = CollectionProcessor(
+            input_channels=input_channels,
+            output_labels=output_labels,
+            processing_fn=processing_fn,
+            sample_size=sample_size,
+            bound=bound,
+            verbose=True,
+            bound_mode=bound_mode
+        )
+
+    labels_bc = cp.process(bc, path)
+
+    in_img = bc.numpy()
+    out_img = labels_bc.numpy()
+
+    if bound_mode == 'drop' or bound_mode is None:
+        assert out_img.dtype == 'uint8'
+    elif bound_mode == 'weight':
+        assert out_img.dtype == 'float32'
+    assert in_img.shape == out_img.shape
+    assert np.max(np.abs(in_img - out_img)) <= 1  # due to rounding can be up to 1
+    shutil.rmtree(path)
+
+
+def test_collection_processor():
```

Comment:
I would prefer non-random tests, because we do not want probability, we want it work every time. It is better to stick to some arbitrary values, because we can at least reproduce it when it fails :)

Also, we can use pytest.parametrize for multiple calls of the same test with different size/dtype/etc.

Reply:
fixed https://github.com/Geoalert/aeronetlib/pull/44/commits/9fa54dca06f356dfd056e89fcf10a2aedddb192d